tests/testthat/test-rcpp-population-interface.R

```diff
@@ -1,39 +1,140 @@
-  }
-
-  clear()
-})
+# Instructions ----
+#' This file follows the format generated by FIMS:::use_testthat_template().
+#' Necessary tests include input and output (IO) correctness [IO
+#' correctness], edge-case handling [Edge handling], and built-in errors and
+#' warnings [Error handling]. See `?FIMS:::use_testthat_template` for more
+#' information. Every test should have a @description tag that takes up just
+#' one line, which will be used in the bookdown report of {testthat} results.
+
+# test_rcpp_population_interface ----
+## IO correctness ----
+test_that("test_rcpp_population_interface() works with correct inputs", {
+  # setup population to create test values
+  population <- methods::new(Population)
+  nyears <- 10
+  nages <- 10
+  log_M_length <- nyears * nages
+  population$log_M$resize(log_M_length)
+  purrr::walk(
+    seq_along(1:log_M_length),
+    \(x) population$log_M[x]$value <- -1
+  )
+  population$log_init_naa$resize(nages)
+  purrr::walk(
+    seq_along(1:nages),
+    \(x) population$log_init_naa[x]$value <- 0
+  )
+  population$log_init_naa$set_all_estimable(TRUE)
+  population$nages$set(nages)
+  population$ages$resize(nages)
+  purrr::walk(
+    seq_along(1:nages),
+    \(x) population$ages$set(x - 1, x)
+  )
+  population$nfleets$set(2)
+  population$nseasons$set(1)
+  population$nyears$set(nyears)
+
+
+  #' @description Test that the population id is 1.
+  expect_equal(
+    object = population$get_id(),
+    expected = 1
+  )
+
+  #' @description Test that the log_M values are all -1.
+  #' @description Test that the log_M values are not estimated.
+  for (i in 1:(nyears * nages)) {
+    expect_equal(
+      object = population$log_M[i]$value,
+      expected = -1
+    )
+    expect_false(
+      object = population$log_M[i]$estimated
+    )
+  }
+
+  #' @description Test that the log_init_naa values are all 0.
+  #' @description Test that the log_init_naa values are estimated.
+  for (i in 1:nyears) {
+    expect_equal(
+      object = population$log_init_naa[i]$value,
+      expected = 0
+    )
+    expect_true(
+      object = population$log_init_naa[i]$estimated
+    )
+  }
+
+  population$log_init_naa$set_all_estimable(FALSE)
+  population$log_M$set_all_estimable(TRUE)
+
+  #' @description Test that the log_M values are now estimated.
+  for (i in 1:(nyears * nages)) {
+    expect_equal(
+      object = population$log_M[i]$value,
+      expected = -1
+    )
+    expect_true(
+      object = population$log_M[i]$estimated
+    )
+  }
+
+  #' @description Test that the log_init_naa values are all 0.
+  #' @description Test that the log_init_naa values not estimated.
+  for (i in 1:nyears) {
+    expect_equal(
+      object = population$log_init_naa[i]$value,
+      expected = 0
+    )
+    expect_false(
+      object = population$log_init_naa[i]$estimated
+    )
+  }
+
+  clear()
+})
+
+## Edge handling ----
+test_that("test_rcpp_population_interface() returns correct outputs for edge cases", {
```

Comment:
Feel free to remove lines 99 through 104 and include a comment saying no edge case tests for now.

Reply:
done